storage/src/index/storage.rs

```diff
@@ -163,41 +196,37 @@ impl<K, V> RemoveValueIterator<'_, K, V> {
     /// Remove the value last returned from this iterator from the map. If no value has been
     /// returned yet, or if the last returned value was removed already, then this is a no-op.
     pub fn remove(&mut self) {
-        if !self.can_remove {
-            return;
-        }
-        self.can_remove = false;
-        self.pruned_counter.inc();
-
-        if let Some(prev_prev) = self.prev_prev {
-            unsafe {
-                (*prev_prev).next = (*self.prev.unwrap()).next.take();
```

Comment:
This is the main bug with remove. prev_prev now is just "prev" , so another remove call does not work as expected.